File: python/1241.py
n = int(input())

for i in range(n):
    a, b = input().split()
    flag = False
    
    if len(b) > len(a):
        flag = True
    
    if flag == False:
        for i in range(1, len(b) + 1):
            if b[-i] != a[-i]:
                flag = True
                break
        
    if flag == False:
        print('encaixa')
    else:
        print('nao encaixa')

File: lua/1241.lua
local n = tonumber(io.read())

for i=1,n do
    local entry = io.read()
    local temp = {}
    for v in string.gmatch(entry, '%S+') do
        table.insert(temp, v)
    end
    local a = temp[1]
    local b = temp[2]
    if string.sub(a, #a - #b + 1, #a) == b then
        print('encaixa')
    else
        print('nao encaixa')
    end
end